### PR TITLE
NAS-128566 / 25.04 / Always display the `Lock` button on the encryption card

### PR DESCRIPTION
**Pull request by bvasilenko**
**Changes:**

The purpose of the task is to add a disabled **Lock** button with a message for datasets created with a key, so that the user does not have questions about what is happening.

**Testing:**

Go to **Datasets > (Choose a dataset)** and try to click **Lock** button

**Expected result:** warning "This dataset cannot be locked"

### CODE REVIEW

File: src/app/pages/datasets/modules/encryption/components/zfs-encryption-card/zfs-encryption-card.component.html [hunk 1]
```diff
@@ -79,11 +79,15 @@ <h3 mat-card-title>{{ 'ZFS Encryption' | translate }}</h3>
...
       <ng-container *ixRequiresRoles="[Role.FullAdmin]">
-        @if (hasPassphrase) {
-          <button mat-button ixTest="lock" (click)="onLock()">
-            {{ 'Lock' | translate }}
-          </button>
-        }
+        <button
+          mat-button
+          ixTest="lock"
+          [matTooltip]="!hasPassphrase ? ('This dataset cannot be locked' | translate) : ''"
```
Comment (undsoft) [inline src/app/pages/datasets/modules/encryption/components/zfs-encryption-card/zfs-encryption-card.component.html:85]: Let's add full tooltip as described in the ticket.

**Review by undsoft — APPROVED** [review]
The disabled button is there.

**Comment by bvasilenko** [comment]
backport

**Comment by bvasilenko** [comment]
backport

**Pull request merged**